### PR TITLE
QueueBasedRouting: Use Running Count as fallback logic

### PR DESCRIPTION
**Pull request by akhurana001**
Changes: 
- Use running count as a fallback if queue count are the same. This doesn't change the core algo, just passes in running-counts if queuecounts are the same for a routing group.
- Some other minor changes/clean-up/version fixes to resolve some build errors.

### CODE REVIEW

**Review by riteshvaryani — APPROVED** [review]
lgtm

**Review by charlessong-lyft — APPROVED** [review]
LGTM

File: gateway-ha/src/main/java/com/lyft/data/gateway/ha/router/PrestoQueueLengthRoutingTable.java [hunk 1]
```diff
@@ -208,25 +211,34 @@ public void updateRoutingTable(String routingGroup, Set<String> backends) {
   /**
    * Update routing Table with new Queue Lengths.
    */
-  public void updateRoutingTable(Map<String, Map<String, Integer>> updatedQueueLengthMap) {
+  public void updateRoutingTable(Map<String, Map<String, Integer>> updatedQueueLengthMap,
```
Comment (charlessong-lyft) [inline gateway-ha/src/main/java/com/lyft/data/gateway/ha/router/PrestoQueueLengthRoutingTable.java:214]: where is this function used?

**Review by charlessong-lyft — COMMENTED** [review]
shall we udpate getEligibleBackEnd function to make use of the new function? Also, can we apply the same strategy for the fall back function provideAdhocBackend, if there is no routing group provided

**Comment by charlessong-lyft** [comment]
chatted offline, LGTM now!

**Review by riteshvaryani — APPROVED** [review]

**Pull request merged**